### DISCUSSION TITLE
Kustomization allowed filename patterns documentation

### DISCUSSION THREAD

**Discussion by v1nsai** [fluxcd/flux2#4908] [Q&A]
I spent several hours yesterday troubleshooting an issue with what felt like an "invisible" file in my [cloudflare deployment located here](https://github.com/v1nsai/homelab/tree/cac766ee6b5499fac28b3e29cef600f4b29bb307/projects/cloudflare).

When the file was named`cloudflare/app/sealed-secret-values.yaml`, I got an error about the file not being found.  Full error here for reference:
`kustomize build failed: accumulating resources: accumulation err='accumulating resources from './cloudflare/app': read /tmp/kustomization-3853049343/projects/cloudflare/app: is a directory': recursed accumulation of path '/tmp/kustomization-3853049343/projects/cloudflare/app': accumulating resources: accumulation err='accumulating resources from 'sealed-secrets.yaml': open /tmp/kustomization-3853049343/projects/cloudflare/app/sealed-secrets.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/tmp/kustomization-3853049343/projects/cloudflare/app/sealed-secrets.yaml' : lstat /tmp/kustomization-3853049343/projects/cloudflare/app/sealed-secrets.yaml: no such file or directory`

After hours of struggling with it, I renamed the file to `sealed-secrets.yaml` and updated the `kustomization.yaml` and suddenly the file was visible and everything worked as expected.  I fiddled with it a bit more and discovered that the filename `sealedsecret-values.yaml` was also invalid and made the file "invisible".

Is there documentation on allowed filenames somewhere that I've missed?  It might be a good idea to document these filename requirements in the official docs and/or in the error message.

I'm looking through the source code for `kustomize.config.k8s.io/v1beta1.kustomization` and `kustomize.toolkit.fluxcd.io/v1.kustomization` but I don't speak `golang` very well so it might take me a while to drill down to these filename restrictions.

**Comment by stefanprodan** [comment]
Here is the documentation for the default ignore patterns: https://fluxcd.io/flux/components/source/gitrepositories/#default-exclusions
